lib/routes/charge/items/item_page.dart

```diff
@@ -246,29 +261,48 @@ class CreateItemPageState extends State<CreateItemPage> {
         backgroundColor: Theme.of(context).canvasColor,
         leading: backBtn.BackButton(),
         title: Text(
-          "Add Item",
+          "$_titlePrefix Item",
           style: Theme.of(context).appBarTheme.textTheme.title,
         ),
         actions: actions == null ? <Widget>[] : actions,
         elevation: 0.0,
       ),
       body: body,
       bottomNavigationBar: SingleButtonBottomBar(
-        text: "Add Item",
+        text: "$_titlePrefix Item",
         onPressed: () {
           {
             if (_formKey.currentState.validate()) {
-              AddItem addItem = AddItem(
-                Item(
-                  name: _nameController.text.trimRight(),
-                  currency: _currencySymbol(),
-                  price: double.parse(_formattedPrice()),
-                ),
-              );
-              widget._posCatalogBloc.actionsSink.add(addItem);
-              addItem.future.then((_) {
-                Navigator.pop(context);
-              });
+              if (_inEditMode) {
+                UpdateItem updateItem = UpdateItem(
+                  Item(
+                    id: widget.item.id,
+                    name: _nameController.text.trimRight(),
+                    currency: _selectedCurrency.shortName,
+                    price: double.parse(
+                        _formattedPrice(double.parse(_priceController.text))),
```

Comment:
why the roundtrip? Can we just do `double.parse` ?
same bellow?

Reply:
[Done](https://github.com/breez/breezmobile/pull/296/commits/fc5073de055417289cde5536d54b7c0e7c1b4b67)